vb-fe/src/components/player/PlayerVolume.tsx

```diff
@@ -14,41 +14,43 @@ import {
   IconButton,
 } from "@chakra-ui/react";
 import { HiVolumeOff, HiVolumeUp } from "react-icons/hi";
+import { useSpotifyPlayer } from "core/player/index";
+
 import { useDebounce } from "util/Input";
-import { WebPlayer } from "core/player/Player";
 
 export const PlayerVolume: FunctionComponent = (): JSX.Element => {
+  const player = useSpotifyPlayer();
   const [debounce] = useDebounce();
-  const [volume, setVolume] = useState<number | null>(null);
+  const [vol, setVolume] = useState<number | null>(null);
 
   useEffect(() => {
-    WebPlayer.getPlayer()
-      .getVolume()
-      .then((vol: number) => {
-        setVolume(vol * 100);
+    if (player) {
+      player.getVolume().then((v: number) => {
+        setVolume(v * 100);
       });
-  }, []);
+    }
+  }, [player]);
 
   const onVolumeChange = (value: number) => {
-    debounce(value, (vol: any) => {
-      setVolume(vol);
-      WebPlayer.getPlayer().setVolume(vol / 100);
+    debounce(value, (v: any) => {
```

Comment:
We're passing `value` into this function, not `v` - does this by chance need to be `value`?

Reply:
So value and v will be the same, but for some reason while using it, it wasn't. `value` was the previous value and `v` was the update value, which is why I had to do that. The debounce callback gets the updated value